design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+<td><a href="https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm?tocpath=Integrations%7COpenShift%252C%20Kubernetes%252C%20and%20GKE%7CDeploy%20Applications%7C_____3">link</a></td>
+</tr>
+<tr class="even">
+<td>Secrets Provider application-mode Helm chart manifest</td>
+<td><a href="https://github.com/cyberark/secrets-provider-for-k8s/blob/master/helm/secrets-provider/templates/secrets-provider.yaml">link</a></td>
+</tr>
+</tbody>
+</table>
+
+## Background
+
+### Secrets Provider Push to File Support
+
+The goal of the Milestone 1 (M1) Push-to-File initiative is to enhance the
+Secrets Provider init container integration with an option for writing
+secrets values that have been retrieved from Conjur into files that can be
+directly accessed by an application that is running in the same Pod.
+
+The intent is to make it much easier for developers to integrate their
+applications directly with Conjur, without having to make significant changes
+or additions to those applications or their containers; that is, without
+having to add Summon to the application container or integrate a Conjur
+client API into the application itself.
+
+### Use of Kubernetes Annotations
+
+The M1 Push-to-File feature requires a flexible way to allow developers or
+deployers to map secrets that are to be retrieved from Conjur to specific
+target file location(s). This will allow applications to consume secrets
+from files at expected locations, further minimizing changes to the
+application.
+
+The M1 Push-to-File feature makes use of
+[Kubernetes Pod Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+to provide a method of flexibly configuring the mapping of secrets to desired
+target locations.
+
+The M1 Push-to-File feature also adds support for configuring other
+Secrets Provider container/application settings using Kubernetes annotations
+as an alternative to using Pod environment variable settings. Allowing
+Secrets Provider to be more generally configured via Pod annotations
+provides a uniform mechanism for configuring the Secrets Provider, and should
+lend itself to an easier upgrade/migration to the proposed CyberArk Dynamic
+Sidecar Injector design.
+
+## Solution
+
+### User Experience
+
+#### Annotations for Secrets Provider Container/Application Configuration
+
+The Secrets Provider container/application configuration can be set using
+either Pod annotations or by using Pod environment variable settings
+([legacy configuration](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm#Step4AddtheCyberArkSecretsProviderforKubernetesimagetotheapplicationdeploymentmanifest)).
+If both a Pod Annotation and a Pod environment variable is configured for
+any particular Secrets Provider setting, then the Pod Annotation configuration
+takes precedence.
```

Comment:
I don't think we're planning on adding an env-var mode for "file"-based flows, so this would only be supported for the "k8s"-based flows